bindings/pyroot/pythonizations/test/tfile_context_manager.py

```diff
@@ -0,0 +1,44 @@
+import os
+import unittest
+
+import ROOT
+
+from ROOT import TFile
+
+
+class TFileContextManager(unittest.TestCase):
+    """
+    Test of TFile used as context manager
+    """
+
+    NBINS = 100
+    XMIN = 0
+    XMAX = 100
+
+    def test_write_histogram(self):
+        """
+        Check that an histogram is correctly written to a file within a context
+        manager.
+        """
+        filename = "TFileContextManager_test_write_histogram.root"
+        with TFile(filename, "recreate") as outfile:
+            hout = ROOT.TH1F("h", "h", self.NBINS, self.XMIN, self.XMAX)
+            outfile.WriteObject(hout, "histo")
+
+        # Need to understand better what to check here. For now simple sanity checks
+        self.assertTrue(bool(outfile))  # The TFile object is still there
```

Comment:
Do you need the `bool(outfile)` or just `outfile` would be evaluated as boolean?

Also, here you just want to check that you still have the file, even if closed already? Perhaps you could check that it's closed? https://root.cern.ch/doc/master/classTFile.html#a67dedbe56cfe4792cff78df129718c11

Reply:
Thanks, this is the type of improvement I was looking for for this test :smile: